src/EditorFeatures/CSharpTest/OnTheFlyDocs/OnTheFlyDocsUtilitiesTests.cs

```diff
@@ -110,21 +88,620 @@ struct CustomStruct
+    }
+
+    private static ISymbol GetMethodSymbol(SyntaxTree syntaxTree, SemanticModel semanticModel)
+    {
+        var methodDeclaration = syntaxTree.GetRoot()
             .DescendantNodes()
-            .OfType<InvocationExpressionSyntax>()
+            .OfType<MethodDeclarationSyntax>()
             .First();
 
-        var methodSymbol = semanticModel!.GetSymbolInfo(methodInvocation).Symbol;
+        return semanticModel.GetDeclaredSymbol(methodDeclaration)!;
+    }
+
+    private static async Task TestSymbolContextAsync(
+        string testCode,
+        Func<SyntaxTree, SemanticModel, ISymbol> getSymbol,
+        string? expectedText = null,
+        bool expectAllNull = false,
+        int resultIndex = 0)
+    {
+        var (solution, _, semanticModel, syntaxTree) = await SetupWorkspaceAsync(testCode);
+
+        var symbol = getSymbol(syntaxTree, semanticModel);
+        var result = OnTheFlyDocsUtilities.GetAdditionalOnTheFlyDocsContext(solution, symbol);
+
+        if (expectAllNull)
+        {
+            Assert.True(result.All(item => item == null));
+            return;
+        }
+
+        var relevantInfo = result[resultIndex];
+        Assert.NotNull(relevantInfo);
+
+        if (expectedText != null)
+        {
+            await AssertSymbolTextMatchesAsync(relevantInfo!, expectedText);
+        }
+    }
+
+    private static Task TestMethodContextAsync(string testCode, string? expectedText = null, bool expectAllNull = false, int resultIndex = 0)
+    {
+        return TestSymbolContextAsync(
+            testCode,
+            (syntaxTree, semanticModel) => GetMethodSymbol(syntaxTree, semanticModel),
+            expectedText,
+            expectAllNull,
+            resultIndex);
+    }
+
+    private static Task TestObjectCreationTypeContextAsync(string testCode, string expectedText)
+    {
+        return TestSymbolContextAsync(
+            testCode,
+            (syntaxTree, semanticModel) =>
+            {
+                var objectCreation = syntaxTree.GetRoot()
+                    .DescendantNodes()
+                    .OfType<ObjectCreationExpressionSyntax>()
+                    .First();
+
+                return semanticModel.GetTypeInfo(objectCreation).Type!;
+            },
+            expectedText);
+    }
+
+    private static Task TestInvocationContextAsync(string testCode, string? expectedText = null, bool expectAllNull = false, int resultIndex = 0)
+    {
+        return TestSymbolContextAsync(
+            testCode,
+            (syntaxTree, semanticModel) =>
+            {
+                var methodInvocation = syntaxTree.GetRoot()
+                    .DescendantNodes()
+                    .OfType<InvocationExpressionSyntax>()
+                    .First();
+
+                return semanticModel.GetSymbolInfo(methodInvocation).Symbol!;
+            },
+            expectedText,
+            expectAllNull,
+            resultIndex);
+    }
```

Comment:
can create single helper fort hsi and the above method that generically is passed a `<TSyntaxNode>` type arg.